### PR TITLE
Graph interaction via Chart.JS

### LINKED ISSUE #153: activeUnit should be renamed to something more descriptive
like activeFilter or dateRange or something

### PR DESCRIPTION
**Pull request by AKaudinov**
- Implemented graph Interaction via Chart.js and chartjs plugin zoom
   - Individual graphs can be zoomed and panned
- Created a new component: `LineGraph` to keep things readable and easier to navigate
   - New component contains the Chart.js line graph, its axis options -- including zooming and panning
- Refactored the building out of the x axes series from each individual graph, into the main graphs component, so the x series gets built only when data changes, or the date filter is changed, and is passed down to each graph, instead of each graph re-computing the same exact thing 10 times.
- Added the ability to overlay the graph on the same chart by enabling a secondary Y axes on the same component, instead of rendering A new graph component.
  - This also keeps the zoom and pan of the graph that the overlay -- secondary data -- will be on top of without re-rendering the component but updating its visibility options, in place
- The graph also formats the the labels to be more precise when `day` filter is selected, to avoid showing the same exact day throughout the whole graph as it does today.

- Switched the `Graphs` component to be lazy loaded (dynamically loaded with no SSR) as Chart.js and zoom depend on Browser window API.

- Refactored out no longer necessary Nivo graph options
- Removed Nivo related packages

- Refactored no longer necessary Graph seeker styles that were previously used to make the graph seeker work on the data page
- Refactored no longer necessary conditional functionality and props in the DateRangePicker component as it's no longer used as a date seeker

---
Other:

- Refactored the progress bar animation to use compositor only properties (e.g. Transform and Opacity) for the animation. This avoids repaints
  which has much less load on the CPU, hence increasing
  performance

---

Also closes #153

### CODE REVIEW

File: components/DataRangePicker/index.js [hunk 1]
```diff
@@ -101,10 +102,12 @@ class DataRangePicker extends React.Component {
             })}
           />
         </div>
+        {(
```
Comment (jedahan) [inline components/DataRangePicker/index.js]: some debugging leftover?

Comment (AKaudinov) [reply components/DataRangePicker/index.js]: Ah, thanks for catching that! forgot to remove those parentheses after removing the conditional.

File: components/Graph/index.js [hunk 2]
```diff
@@ -135,25 +74,17 @@ const Graph = ({
       <div className={styles.graphWrapper}>
         <LineGraph
           {...graph}
+          activeUnit={activeUnit}
+          xSeries={xSeries}
+          data={graph.data}
+          domain={[graph.domain[0], graph.domain[1]]}
```
Comment (jedahan) [inline components/Graph/index.js]: will be removed

File: components/DataRangePicker/index.js [hunk 3]
```diff
@@ -85,8 +85,9 @@ class DataRangePicker extends React.Component {
 
     return (
       <div className={styles.container}>
-        <div className={styles.sliderContainer}>
+        <div className={`${styles.sliderContainer} ${styles.customSliderContainer}`}>
```
Comment (jedahan) [inline components/DataRangePicker/index.js]: so we can `[styles.sliderContainer, styles.customSliderContainer]`

**Review by jedahan — APPROVED** [review]

**Pull request merged**